server/aws-lsp-codewhisperer/src/language-server/inline-completion/codeWhispererServer.ts

```diff
@@ -630,22 +630,14 @@ export const CodewhispererServerFactory =
                     if (session.suggestionType === SuggestionType.EDIT && acceptedSuggestion.content) {
                         // [acceptedSuggestion.insertText] will be undefined for NEP suggestion. Use [acceptedSuggestion.content] instead.
                         // Since [acceptedSuggestion.content] is in the form of a diff, transform the content into addedCharacters and deletedCharacters.
-                        const addedAndDeletedChars = getAddedAndDeletedChars(acceptedSuggestion.content)
-                        if (addedAndDeletedChars) {
-                            addedCharactersForEditSuggestion = addedAndDeletedChars.addedCharacters
-                            deletedCharactersForEditSuggestion = addedAndDeletedChars.deletedCharacters
-
-                            codePercentageTracker.countAcceptedTokens(
-                                session.language,
-                                addedCharactersForEditSuggestion
-                            )
-                            codePercentageTracker.countTotalTokens(
-                                session.language,
-                                addedCharactersForEditSuggestion,
-                                true
-                            )
-                            enqueueCodeDiffEntry(session, acceptedSuggestion, addedCharactersForEditSuggestion)
-                        }
+                        const { addedLines, deletedLines } = getAddedAndDeletedLines(acceptedSuggestion.content)
+                        const charDiffResult = getCharacterDifferences(addedLines, deletedLines)
+                        addedCharactersForEditSuggestion = 'a'.repeat(charDiffResult.charactersAdded)
+                        deletedCharactersForEditSuggestion = 'a'.repeat(charDiffResult.charactersRemoved)
```

Comment:
should we just pass the length of it instead of a fake string

Reply:
I have though about that. The current countToken() only takes strings since it's inherited from inline suggestions. We will add the countToken() that takes a number later because there are some non-trivial logic in especially countTotalTokens.